### PR TITLE
add albumid option to generate command, fix #41

### LINKED ISSUE #41: Support for generation of specific albums (like sw:thumbnail:generate)

### PR DESCRIPTION
**Pull request by larsbo**
To generate webp images only for a specific album like the `sw:thumbnail:generate` command.

### CODE REVIEW

File: Commands/GenerateWebpImages.php [hunk 1]
```diff
@@ -60,4 +68,22 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+     * @param int $albumId
+     *
+     * @return array
+     */
+    protected function getMedia($albumId)
+    {
+        $connection = $this->container->get('dbal_connection');
+
+        if (empty($albumId)) {
+            return $connection->fetchAll('SELECT * FROM s_media WHERE type = "IMAGE"');
```
Comment (shyim) [inline Commands/GenerateWebpImages.php]: can you switch to querybuilder :sweat_smile:

File: Commands/GenerateWebpImages.php [hunk 2]
```diff
@@ -28,36 +61,155 @@ protected function configure()
...
+
+        return $builder->getQuery()->getResult();
+    }
+
+    /**
+     * @param Album $album
+     */
+    private function createAlbumThumbnails(Album $album)
+    {
+        $this->output->writeln("Generating WebP images for Album {$album->getName()} (ID: {$album->getId()})");
```
Comment (shyim) [inline Commands/GenerateWebpImages.php]: Please use sprintf instead :)

File: Commands/GenerateWebpImages.php [hunk 3]
```diff
@@ -28,36 +61,155 @@ protected function configure()
...
+     * @throws \Exception
+     *
+     * @return bool
+     */
+    private function imageExists(Media $media)
+    {
+        $mediaService = $this->container->get('shopware_media.media_service');
+        $projectDir = $this->container->getParameter('shopware.app.rootdir');
+
+        return $mediaService->has($projectDir . $media->getPath());
```
Comment (shyim) [inline Commands/GenerateWebpImages.php]: Why preprending $projectDir?

Comment (larsbo) [reply Commands/GenerateWebpImages.php]: 🤔 copied from Shopware thumbnail generation command.

Comment (shyim) [reply Commands/GenerateWebpImages.php]: $media->getPath should be enought

**Comment by larsbo** [comment]
Also excluded recycle bin album from image generation 😉

**Comment by shyim** [comment]
Fixed with #45

**Pull request closed without merging**